src_base/xeve_args.c

```diff
@@ -0,0 +1,2271 @@
+{
+    int i, ret = 0, idx = 0;
+    const char* fname_cfg = NULL;
+    FILE* fp;
+
+    int num_configs = 0;
+    int pos_conf_files[ARGS_MAX_NUM_CONF_FILES];
+
+     // do some checks
+    if(!args || argc<=0 || !argv)
+        return XEVE_ERR_INVALID_ARGUMENT;
+
+    if( !args->initialized )
+        return XEVE_ERR;
+
+    memset(&pos_conf_files, -1, sizeof(int) * ARGS_MAX_NUM_CONF_FILES);
+
+    /* config file parsing */
+    for (i = 1; i < argc; i++)
+    {
+        if (!strcmp(argv[i], "--"ARGS_KEY_LONG_CONFIG))
+        {
+            if (i + 1 < argc)
+            {
+                num_configs++;
+                pos_conf_files[num_configs - 1] = i + 1;
+            }
+        }
+    }
+    for (int i = 0; i < num_configs; i++)
+    {
+        fname_cfg = argv[pos_conf_files[i]];
+        if (fname_cfg)
+        {
+            fp = fopen(fname_cfg, "r");
+            if (fp == NULL) return -1; /* config file error */
+
+            if (args_parse_cfg(fp, args->opts, 1))
+            {
+                fclose(fp);
+                return -1; /* config file error */
+            }
+            fclose(fp);
+        }
+    }
+    /* command line parsing */
+    while (1)
+    {
+        ret = args_parse_cmd(argc, argv, args->opts, &idx, errstr);
+        if (ret <= 0) break;
+    }
+    return ret;
+}
+
+int xeve_args_check_mandatory(ARGS_PARSER * args, char ** err_arg)
+{
+    // do some checks
+    if(!args)
+        return XEVE_ERR_INVALID_ARGUMENT;
+
+    ARGS_OPT * o = args->opts;
+
+    while (o->key != 0)
+    {
+        if (o->is_mandatory)
+        {
+            if (o->is_set == 0)
+            {
+                /* not filled all mandatory argument */
+                *err_arg = o->key_long;
+                return XEVE_ERR;
+            }
+        }
+        o++;
+    }
+    return XEVE_OK;
+}
+
+int XEVE_EXPORT xeve_args_get_option_val(ARGS_PARSER * args, char* keyl,  char** val) {
+
+    char buf[ARGS_OPT_VALUE_MAXLEN];
+    size_t len;
+    char *str;
+    ARGS_OPT *opt = NULL;
+
+    *val = NULL;
+
+    // do some checks
+    if( !args || !keyl )
+        return XEVE_ERR_INVALID_ARGUMENT;
+
+    int idx = args_search_long_key(args->opts, keyl);
+    if( idx  < 0 )
+        return XEVE_ERR_INVALID_ARGUMENT;
+
+    //if( !args->initialized )
+    //    return XEVE_ERR;
+
+    if(!args->opts[idx].opt_storage)
+        return XEVE_ERR;
+
+    if( args->opts[idx].is_set != 1 )
+        return XEVE_ERR;
+
+    opt = args->opts + idx;
+
+    switch(opt->value_type)
+    {
+        case VAL_TYPE_INT:
+            snprintf(buf, ARGS_OPT_VALUE_MAXLEN, "%d", *((int*)(args->opts[idx].opt_storage)));
+
+            len = strlen(buf);
+            *val = (char*)malloc(len+1);
+            strncpy(*val, buf, len+1);
+            // *val[len-1] = '\0';
+
+            break;
+        case VAL_TYPE_STRING:
+
+            str = (char*)(args->opts[idx].opt_storage);
+            if(!str) return XEVE_ERR;
```

Comment:
Line 1855 already does the check for NULL args->opts[idx].opt_storage, so this if block might be unreachable. please check